public/app/features/dashboard-scene/sharing/public-dashboards/SharePublicDashboardTab.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+
+import { getBackendSrv } from '@grafana/runtime';
+import { SceneComponentProps, SceneObjectBase, SceneObjectRef } from '@grafana/scenes';
+import { t } from 'app/core/internationalization';
+import { Loader } from 'app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard';
+import {
+  PublicDashboard,
+  publicDashboardPersisted,
+} from 'app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboardUtils';
+import { trackDashboardSharingActionPerType } from 'app/features/dashboard/components/ShareModal/analytics';
+import { shareDashboardType } from 'app/features/dashboard/components/ShareModal/utils';
+
+import { DashboardScene } from '../../scene/DashboardScene';
+import { SceneShareTabState } from '../types';
+
+import { ConfigPublicDashboard } from './ConfigPublicDashboard';
+import { CreatePublicDashboard } from './CreatePublicDashboard';
+
+export interface SharePublicDashboardTabState extends SceneShareTabState {
+  dashboardRef: SceneObjectRef<DashboardScene>;
+  isGetLoading: boolean;
+  isUpdateLoading?: boolean;
+  isSaveLoading?: boolean;
+  publicDashboard?: PublicDashboard;
+}
+
+export class SharePublicDashboardTab extends SceneObjectBase<SharePublicDashboardTabState> {
```

Comment:
@kaydelaney  - I wonder if this object couldn't be just a naive wrapper around `SharePublicDashboard`? I see @juanicabanas point of this work not using the RTK API that we already have implemented. 

I think the primary refactor of current components will require a change to how the Public Dashboard modal accesses the dashboard model and related things like variables. Currently, it reads it from the redux, we need to make components like  `ConfigPublicDashboard` dummy and accept the model via props. Similarly we need to remove the dependency of reading variables from DashboardModel and instead get those from the scene hierarchy. So it mostly seems like the refactoring should happen in `CreatePublicDashboard` and `ConfigPublicDashboard` components so that they can be used both in the new modal and the old one.

Reply:
> I think the primary refactor of current components will require a change to how the Public Dashboard modal accesses the dashboard model and related things like variables. Currently, it reads it from the redux, we need to make components like ConfigPublicDashboard dummy and accept the model via props

I think that's what I did with this commit https://github.com/grafana/grafana/pull/76837/commits/219697bdefe9c9e4b9afc87fbb4e7ff8381022aa ? Unless I'm missing something?

My main reason for not using the existing RTK stuff is that within the query methods `dashboard.updateMeta` is used, which I don't think would work for scenes.
I could add some logic to the query methods to handle things differently for scenes/dashboards?

Reply:
> I think that's what I did with this commit https://github.com/grafana/grafana/commit/219697bdefe9c9e4b9afc87fbb4e7ff8381022aa ? Unless I'm missing something

yes, you are right @kaydelaney. 

> My main reason for not using the existing RTK stuff is that within the query methods dashboard.updateMeta is used, which I don't think would work for scenes.


I see this now. That's true, it wouldn't work given it's executed against dashboard json. Is there a scenes equivalent for that implemented in this migration? Or isn't it even needed?

Reply:
the dashboard is taken from `dashboardState.getModel()`, which I think there's an equivalent in scenes, right?
On the other hand, I don't see the point of removing the whole RTK query use for that specific case. 
I mean, it's possible to wait for the RTK query hook to be executed from the component side itself and then, update the dashboard for example.
I think the problem of updating the dashboard that you mentioned is independent of the discussions about implementing RTK query or not.

Reply:
I just saw the changes. I'll take a look !